docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,265 @@ If you need to have external labels in your query results, this is currently not
+
+    For AWS S3, use the aws tool in the following command:
+
+    ```bash
+    aws s3 cp -r s3://<THANOS-BUCKET> s3://<TENANT>/<DIRECTORY>
+    ```
+
+    For Google Cloud Storage (GCS), use the gsutil tool in the following command:
+    
+    ```bash
+    gsutil -m cp -r gs://<THANOS-BUCKET>/* gs://<INTERMEDIATE-MIMIR-BUCKET>/
+    ```
+    
+    Once the copy process is done, inspect the blocks in the bucket to ensure it's valid from Thanos perspective.
+    
+    ```bash
+    thanos tools bucket inspect \
+        --objstore.config-file bucket.yaml
+    ```
+
+2) Remove the downsampled blocks
+
+   Mimir doesn't understand the downsampled blocks(ie. blocks with non-zero Resolution field in meta.json file) in Thanos. Therefore we need to remove the 5m and 1h downsampled blocks from this bucket.
+   
+   Mark the downsampled blocks for deletion.
+   
+   ```bash
+   thanos tools bucket retention \
+       --objstore.config-file bucket.yaml \
+       --retention.resolution-1h=2h \
+       --retention.resolution-5m=2h \
+       --retention.resolution-raw=0d
+   ```
+   Cleanup the blocks marked for deletion.
+   
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+3) Remove the duplicated blocks
+
+   If two replicas of Prometheus instancesare deployed for HA, then only upload the blocks from one of the replicas and drop the other.
+   
+   ```bash
+   # Get list of all blocks in the bucket
+   thanos tools bucket inspect \
+       --objstore.config-file bucket.yaml \
+       --output=tsv >> blocks.tsv
+       
+   # Find blocks from replica that we will drop    
+   cat blocks.tsv| grep prometheus_replica=<PROMETHEUS-REPLICA-TO-DROP> \
+       | awk '{print $1}' >> blocks_to_drop.tsv
+       
+   # Mark found blocks for deletion
+   for ID in $(cat blocks_to_drop.tsv)
+   do
+       thanos tools bucket mark \
+          --marker="deletion-mark.json" \
+          --objstore.config-file bucket.yaml \
+          --details="Removed as duplicate" \
+          --id $ID
+   done
+   ```
+   
+   > **Note**: Replace **prometheus_replica** with the unique label that would differentiate prometheus replicas in your setup.
+
+   Again cleanup the duplicate blocks marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **ProTip!**: If you want to visualize what exactly is happening in the blocks with respect to the source of blocks, labels, compaction levels, etc, you can use the below command to get the output as CSV and import it to a google sheet.
+   > ```bash 
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket-prod.yaml \
+   >     --output=csv >> thanos-blocks.csv
+   > ```
+
+4) Relabel the blocks with external labels
+
+   Relabel the blocks with the required external labels. Create a rewrite config like the following:
```

Comment:
We should point out what labels to use: basically external labels found in the blocks.

How does one check which external labels block have, and whether they are the same across all blocks?

For example something like this can show which labels are used, and how many different combinations are there:

```
for FILE in $(gsutil ls 'gs://<bucket>/meta.json'); do
  gsutil cat $FILE | jq --compact-output --sort-keys .thanos.labels
done | sort | uniq -c
```

Reply:
Just before step 4, I added a protip block where we export the details of the blocks as CSV. If we import the CSV into a google sheet, we will be able to see all the labels available and we can apply filters to get the different combinations as well.  Wouldn't that be useful?